target_snowflake/__init__.py

```diff
@@ -130,45 +135,47 @@ def persist_lines(config, lines, information_schema_cache=None):
                 validators[stream].validate(float_to_decimal(o['record']))
             except Exception as ex:
                 if type(ex).__name__ == "InvalidOperation":
-                    logger.error("Data validation failed and cannot load to destination. RECORD: {}\n'multipleOf' validations that allows long precisions are not supported (i.e. with 15 digits or more). Try removing 'multipleOf' methods from JSON schema."
-                    .format(o['record']))
+                    logger.error(
+                        "Data validation failed and cannot load to destination. RECORD: {}\n'multipleOf' validations that allows long precisions are not supported (i.e. with 15 digits or more). Try removing 'multipleOf' methods from JSON schema."
+                            .format(o['record']))
                     raise ex
 
-            primary_key_string = stream_to_sync[stream].record_primary_key_string(o['record'])
-            if not primary_key_string:
-                primary_key_string = 'RID-{}'.format(row_count[stream])
-
             if stream not in records_to_load:
-                records_to_load[stream] = {}
+                records_to_load[stream] = []
 
+            # append record
             if config.get('add_metadata_columns') or config.get('hard_delete'):
-                records_to_load[stream][primary_key_string] = add_metadata_values_to_record(o, stream_to_sync[stream])
```

Comment:
is this removed for purpose? what if we receive multiple version of the same record in the stream?
For example when a record was updated we need to deduplicate and keep only the latest version of the record identified by PK?

Reply:
Yeah, I removed it on purpose. I figured Snowflake [merge_sql query](https://github.com/transferwise/pipelinewise-target-snowflake/blob/b104b9712ae039b2747ec31368364199acdc26c5/target_snowflake/db_sync.py#L451) does take care of updating same record.

Reply:
I have run tests with and without those sub-routines and the behavior was the same.
However, now that you explained why we update the dictionary. I'll put back that piece of code.

Reply:
thank you, I've just realised there are no test cases for de-duplicating events. I'll try to add some